File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxprofit=0;
        int min=Integer.MAX_VALUE;
        for(int price:prices){
            min=Math.min(min,price);
            maxprofit=Math.max(maxprofit,price-min);
        }
        return maxprofit;
    }

}

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        min=prices[0]
        maxprofit=0
        for i in range(len(prices)):
            if prices[i]<min:
                min=prices[i]
            elif prices[i] - min >maxprofit:
                maxprofit=prices[i]-min
                

        return maxprofit

              



        